NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -123,136 +636,382 @@ def calculate_simulation_weights(positions, zenith, azimuth, site='summit', debu
+
+        Parameters
+        ----------
+        interp_lowfreq : float
+            lower frequency for the bandpass filter in interpolation in GHz
+        interp_highfreq : float
+            higher frequency for the bandpass filter in interpolation in GHz
+
+        Returns
+        -------
+        efield_interpolator : interpolator object
+
+        """
+        self.efield_interpolator_initialized = True
+        self.interp_lowfreq = interp_lowfreq
+        self.interp_highfreq = interp_highfreq
+
+        geomagnetic_angle = get_geomagnetic_angle(self.zenith, self.azimuth, self.shower[shp.magnetic_field_vector])
+
+        if geomagnetic_angle < 15 * units.deg:
+            logger.warning(
+                f'geomagnetic angle is {geomagnetic_angle / units.deg:.2f} deg, '
+                f'which is smaller than 15deg, which is the lower limit for the signal interpolation. '
+                f'The closest obersever is used instead.')
+            self.efield_interpolator = -1
+        else:
+            logger.info(
+                f'electric field interpolation with lowfreq {interp_lowfreq / units.MHz} MHz '
+                f'and highfreq {interp_highfreq / units.MHz} MHz')
+
+            self.efield_interpolator = cr_pulse_interpolator.signal_interpolation_fourier.interp2d_signal(
+                self.obs_positions_vxB_vxvxB[:, 0],
+                self.obs_positions_vxB_vxvxB[:, 1],
+                self.electric_field_on_sky,
+                signals_start_times=self.efield_times[:, 0],
+                lowfreq=(interp_lowfreq - 0.01) / units.MHz,
+                highfreq=(interp_highfreq + 0.01) / units.MHz,
+                sampling_period=1 / self.sampling_rate / units.s,  # interpolator wants sampling period in seconds
+                phase_method="phasor",
+                radial_method='cubic',
+                upsample_factor=5,
+                coherency_cutoff_threshold=0.9,
+                ignore_cutoff_freq_in_timing=False,
+                verbose=False
+            )
+        return self.efield_interpolator
+
+    def initialize_fluence_interpolator(self, quantity=efp.signal_energy_fluence, debug=False):
+        """
+        initilized fluence interpolator object.
+
+        Parameters
+        ----------
+        quantity : electric field parameter
+            quantity to interpolate, e.g. efp.signal_energy_fluence
+            The quantity needs to be available as parameter in the electric field object!!! You might
+            need to run the electricFieldSignalReconstructor. The default is efp.signal_energy_fluence.
+
+        Returns
+        -------
+        fluence_interpolator : interpolator object
+        """
+        fluence_per_position = [
+            np.sum(efield[quantity]) for efield in self.sim_station.get_electric_fields()
+        ]  # the fluence is calculated per polarization, so we need to sum them up
```

Comment:
There should be an option that this is calculated from the traces?

Reply:
@cg-laser ? I am happy with everything.